app/presenters/tree_builder_storage.rb

```diff
@@ -24,8 +24,13 @@ def x_get_tree_roots(count_only, _options)
   end
 
   def x_get_tree_custom_kids(object, count_only, options)
-    return count_only ? 0 : [] if object[:id] != "global"
-    objects = MiqSearch.where(:db => options[:leaf]).visible_to_all
+    objects = MiqSearch.where(:db => options[:leaf])
+    objects = case object[:id]
```

Comment:
and you can use 
`objects = object[:id] == 'global' ?  objects.visible_to_all : objects.your_new_method_from_comment_below` 

what do you think ?